src/topology/alexandroff.lean

```diff
@@ -0,0 +1,308 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import topology.separation
+import topology.opens
+
+/-!
+# The Alexandroff Compactification
+We construct the Alexandroff compactification of an arbitrary topological space `X` and prove
+some properties inherited from `X`.
+
+## Main defintion
+* `alexandroff`: the Alexandroff compactification
+* `of`: the inclusion map defined by `option.some`. This map requires the argument
+        `topological_space X`
+* `infty`: the extra point
+
+## Main results
+* The topological structure of `alexandroff X`
+* The connectedness of `alexandroff X` for a noncompact, preconnected `X`
+* `alexandroff X` is `T₁` for a T₁ space `X`
+* `alexandroff X` is Hausdorff if `X` is locally compact and Hausdorff
+-/
+
+noncomputable theory
+open set
+open_locale classical topological_space
+
+section basic
+
+/-- The Alexandroff extension of an arbitrary topological space `X` -/
+@[nolint unused_arguments]
+def alexandroff (X : Type*) [topological_space X] := option X
```

Comment:
Why do you need `[topological_space X]` in the definition of `alexandroff`?

Reply:
Thanks for the review. Due to my very limited math background, I have the impression that people would never use `alexandroff` if `X` is not a topological space. But I am not sure if I should define `alexandroff` with this argument.

Reply:
Yes, my question was about Lean definition, not math definition. As you can see, the definition of `stone_cech X` uses the argument `[topological_space X]`. This is the difference between `stone_cech` and `with_top`. In cases like `with_top`/`with_bot`/`with_one`/`alexandroff` we avoid unused arguments.

There is one important case when we allow unused arguments in type synonyms, that is when some structures on the new type use these arguments. E.g., `category_theory.single_obj (M : Type*) := unit` is a one-object category with `Hom(*, *) = M` and `f ≫ g = g * f`.

Reply:
Thank you for the detailed explanation! I will modify the code.